### PR TITLE
Thumbnail update

### PR DESCRIPTION
**Pull request by martikat**
Update to ensure default image is loaded when it has not been set

### CODE REVIEW

**Review by peterdavidhamilton — CHANGES_REQUESTED** [review]
Please implement my suggested refactor and try adding a spec

File: app/helpers/content_helper.rb [hunk 1]
```diff
@@ -53,45 +55,69 @@ def side_navigation(page)
...
   def content_footer
     Resource.by_name('ctas.content_footer') || null_resource('ctas.content_footer')
   end
 
+  # @return [Resource, OpenStruct]
   def other_useful_resources
     Resource.by_name('other_useful_resources') || null_resource('other_useful_resources')
   end
 
+  # @return [OpenStruct]
```
Comment (peterdavidhamilton) [inline app/helpers/content_helper.rb:98]: `@param`

File: app/helpers/content_helper.rb [hunk 2]
```diff
@@ -53,45 +55,69 @@ def side_navigation(page)
...
       name: name,
       title: "Title for #{name}",
       body: "Body for #{name}",
       link_to_text: "Link for #{name}",
       link_to: '#',
     )
   end
+
+  # @return [String]
+  def card_thumbnail(card)
```
Comment (peterdavidhamilton) [inline app/helpers/content_helper.rb:110]: `@param` which might make you decide to rename the variable

File: app/helpers/content_helper.rb [hunk 3]
```diff
@@ -53,45 +55,69 @@ def side_navigation(page)
...
     Resource.by_name('ctas.content_footer') || null_resource('ctas.content_footer')
   end
 
+  # @return [Resource, OpenStruct]
   def other_useful_resources
     Resource.by_name('other_useful_resources') || null_resource('other_useful_resources')
   end
 
+  # @return [OpenStruct]
   def null_resource(name)
```
Comment (peterdavidhamilton) [inline app/helpers/content_helper.rb:99]: We discussed naming consistency of `placeholder_<Object>`, i.e. `placeholder_resource` and `placeholder_asset` or similar.

File: spec/helpers/content_helper_spec.rb [hunk 4]
```diff
@@ -128,4 +128,21 @@
       end
     end
   end
+
+  describe '#card_thumbnail' do
+    let(:null_resource) { helper.null_resource('example') }
```
Comment (peterdavidhamilton) [inline spec/helpers/content_helper_spec.rb:133]: This is a very confusing spec. Does it give us greater confidence in our code doing the right thing?

File: spec/models/page_spec.rb [hunk 5]
```diff
@@ -165,18 +165,20 @@
     end
   end
 
-  describe 'Thumbnail' do
-    let(:slug) { 'reducing-paperwork' }
+  describe '#thumbnail' do
+    context 'when help page with image' do
```
Comment (peterdavidhamilton) [inline spec/models/page_spec.rb:169]: `'when page has image'` or `'with linked asset'`
vs.
`'when page has no image'` or `'without linked asset'`

The first makes for grammatically correct spec docs but the latter is even better.
1. context
2. with or without
3. what specifically

**Review by peterdavidhamilton — CHANGES_REQUESTED** [review]

File: spec/helpers/content_helper_spec.rb [hunk 6]
```diff
@@ -119,13 +119,40 @@
...
   end
+
+  describe '#card_thumbnail' do
+    let(:page) { Page.by_slug(slug) }
+
+    context 'with image' do
+      let(:slug) { 'reducing-paperwork' }
+
+      it do
+        expect(helper.card_thumbnail(page)).to include 'images.ctfassets'
```
Comment (peterdavidhamilton) [inline spec/helpers/content_helper_spec.rb:139]: It'd be good if the asset url specs test similarly, ie a FQDN and protocol

File: spec/helpers/content_helper_spec.rb [hunk 7]
```diff
@@ -119,13 +119,40 @@
...
+      end
+    end
+
+    context 'with placeholder' do
+      let(:slug) { 'interactions' }
+
+      it do
+        expect(helper.card_thumbnail(page)).to include 'https://placehold.co'
```
Comment (peterdavidhamilton) [inline spec/helpers/content_helper_spec.rb:147]: `start_with`

File: app/models/page.rb [hunk 8]
```diff
@@ -55,9 +55,9 @@ def navigation?
     page_style == 'side-nav'
   end
 
-  # @return [ContentfulModel::Asset]
+  # @return [ContentfulModel::Asset, nil]
   def thumbnail
-    return ::OpenStruct.new(url: 'https://placehold.co/380x254/347ca9/FFFFFF/png') if fields[:image].blank?
+    return nil if fields[:image].blank?
```
Comment (peterdavidhamilton) [inline app/models/page.rb:60]: You could remove `nil` if you like and it'd behave the same because it is implicit

**Review by David-Feetenby — APPROVED** [review]

**Pull request merged**